symfit/core/fit.py

```diff
@@ -209,14 +321,20 @@ def __str__(self):
         template = "{}({}; {}) = {}"
         parts = []
         for var, expr in self.items():
+            params_sorted = sorted((x for x in self.connectivity_mapping[var]
+                                    if isinstance(x, Parameter)),
+                                   key=lambda x: x.name)
+            vars_sorted = sorted((x for x in self.connectivity_mapping[var]
+                                  if x not in params_sorted),
+                                 key=lambda x: x.name)
```

Comment:
Why not just take `self.params` and `self.independent_vars + self.interdependent_vars`?

Reply:
That would print every component as a function of all vars and params, but I think it is nicer in light of the new call signatures, where from now on we will only call expressions with the kwargs they actually need, that it is also better to print them with the vars they need.

E.g. `{y: a * x, z: x}` prints as
```python
[y(x; a) = a * x,
 z(x) = x]
```
It might also be nice to directly print from function dict instead, giving
```python
{y(x, a): a * x,
 z(x): x}
```
This might be nicer because it is also essentially valid code, but what I like about the current implementation is that it is more mathematical.

Reply:
Make that the `__repr__` together with a `Model` around it.